test/runtests.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+SCRIPT_DIR=$(dirname "$0")
+PARENT_REPO_DIR=$(git rev-parse --show-superproject-working-tree)
+export UTILITIES_RESOURCE_DIR="$SCRIPT_DIR/../resource"
+export TEST_DATA_DIR="$SCRIPT_DIR/data"
+if [ ! -f "$UTILITIES_RESOURCE_DIR/schema.json" ]; then
+	if [ -f "$PARENT_REPO_DIR/resource/schema/global/schema.json" ]; then
+		cp "$PARENT_REPO_DIR/resource/schema/global/schema.json" "$UTILITIES_RESOURCE_DIR/schema.json"
+	else
+		wget -O "$UTILITIES_RESOURCE_DIR/schema.json" -- https://api.zotero.org/schema
+	fi
+fi
```

Comment:
I don't think this is the right approach.

First, when run outside of the client in a fresh install (e.g., in CI here), it's making test success dependent on the current state of https://api.zotero.org/schema, when there's no guarantee of what version that's going to be at any given time. The test results should be based solely on the code in each commit so that CI is meaningful.

And then for an existing install, copying in an external file only when the local schema.json file doesn't already exist means that the result would never change after the first run, even as the parent repo or the API version did. And the local file is even hidden behind a .gitignore, so it would be very easy to forget it was affecting the results.

It seems like there are two things we want to test:

1) We want to test the code in this repo independently, against a known version of schema.json. That seems like a job for a submodule.

2) We want to confirm that a given version of this repo is compatible with the current schema.json in zotero-client (or other repos where it's included as a submodule). For that, it seems like we want to run these tests along with zotero-client test runs, using the schema.json from that repo.

In both cases, schema.json should be copied to the required location at each runtime, from either the embedded schema submodule for (1) or the zotero-client's schema submodule for (2). For (2), it's probably better to specify the schema.json path on the command line (or an environmental variable) rather than rely on a hard-coded path here, so that other repos (e.g., web-library) can test against their own bundled version. The file should never be downloaded from api.zotero.org.

(And then I don't think we need `UTILITIES_RESOURCE_DIR` — just a way to specify the path to schema.json.)

Reply:
OK, replaced `UTILITIES_RESOURCE_DIR` with a `UTILITIES_SCHEMA_PATH` env variable that can be set with the `runtests.sh` `-j` option. Let me know how it looks now.

Reply:
(And added the submodule, etc)